Code/Source/Lidar/LidarRaycaster.cpp

```diff
@@ -116,9 +114,9 @@ namespace RGL
         m_graph.SetIsPcPublishingEnabled(ShouldEnablePcPublishing());
     }
 
-    ROS2::RaycastResult LidarRaycaster::PerformRaycast(const AZ::Transform& lidarTransform)
+    AZ::Outcome<ROS2::RaycastResults, const char*> LidarRaycaster::PerformRaycast(const AZ::Transform& lidarTransform)
     {
-        AZ_Assert(m_range.has_value(), "Programmer error. Raycaster range not configured.");
+        AZ_Assert(m_range.has_value() && m_raycastResults.has_value(), "Programmer error. Raycaster not fully configured.");
```

Comment:
Consider making two assertions with a more precise error message for each. `Raycaster range not configured` is more helpful in finding the bug than `Raycaster not fully configured`.